cmake/tests.cmake

```diff
@@ -2,39 +2,56 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
-function(register_test)
-  set(TEST_TARGET "${ARGV0}")
+# The register_test function automatically registers a test with CTest.
+# So it will be automatically executed in the CI and a xUnit-based XML
+# report will be generated.
+#
+# Usage examples:
+# - default:
+#  register_test(OrbitQtTests)
+#
+# - with test isolation:
+#  register_test(OrbitQtTests TEST_ISOLATION)
+#
+# - with custom timeout:
+#  register_test(OrbitQtTests TIMEOUT 120)
+#
+# - combined:
+#  register_test(OrbitQtTests TEST_ISOLATION TIMEOUT 120)
+
+function(register_test TEST_TARGET)
+  cmake_parse_arguments(ARGS "TEST_ISOLATION" "TIMEOUT" "" ${ARGN})
   string(REGEX REPLACE "Tests$" "" TEST_NAME "${TEST_TARGET}")
 
   set(TESTRESULTS_DIRECTORY "${CMAKE_BINARY_DIR}/testresults")
+
+  if(ARGS_TEST_ISOLATION)
+    set(ISOLATE "--isolate")
```

Comment:
is there such a parameter?

Reply:
Yes. Even though it's the default. So we can also omit it.